test/quantization/test_quantize.py

```diff
@@ -422,35 +422,33 @@ def test_save_load_state_dict(self):
                 out = model(x)
                 self.assertEqual(ref, out)
 
+    @skipIfNoFBGEMM
     def test_activations(self):
         r"""
         Test quantization of activations
         """
-        if 'fbgemm' in supported_qengines:
-            model = ActivationsTestModel()
-            model.qconfig = torch.quantization.get_default_qconfig('fbgemm')
-            prepare(model, inplace=True)
-            self.checkObservers(model)
-            test_only_eval_fn(model, self.calib_data)
-            model = convert(model)
+        model = ActivationsTestModel()
+        model.qconfig = torch.quantization.get_default_qconfig('fbgemm')
+        prepare(model, inplace=True)
+        self.checkObservers(model)
+        test_only_eval_fn(model, self.calib_data)
+        model = convert(model)
 
-            def checkQuantized(model):
-                self.checkNoPrepModules(model.hardswish)
-                self.assertEqual(type(model.hardswish), nnq.Hardswish)
-                test_only_eval_fn(model, self.calib_data)
-                self.checkScriptable(model, self.calib_data)
+        def checkQuantized(model):
+            self.checkNoPrepModules(model.hardswish)
+            self.assertEqual(type(model.hardswish), nnq.Hardswish)
+            test_only_eval_fn(model, self.calib_data)
+            self.checkScriptable(model, self.calib_data)
 
-            checkQuantized(model)
+        checkQuantized(model)
 
-            # test one line API
-            model_oneline = quantize(ActivationsTestModel(), test_only_eval_fn,
-                                     self.calib_data)
-            checkQuantized(model_oneline)
+        # test one line API
+        model_oneline = quantize(ActivationsTestModel(), test_only_eval_fn,
+                                 self.calib_data)
+        checkQuantized(model_oneline)
 
 
-@unittest.skipUnless('fbgemm' in torch.backends.quantized.supported_engines,
-                     " Quantized operations require FBGEMM. FBGEMM is only optimized for CPUs"
-                     " with instruction set support avx2 or newer.")
+@skipIfNoFBGEMM
```

Comment:
@malfet seems like this doesn't work when you decorate class test with it. 
I ran `python test/test_quantization.py TestPostTrainingDynamic`

See the following error
```
Traceback (most recent call last):
  File "test/test_quantization.py", line 62, in <module>
    run_tests()
  File "/home/supriyar/local/miniconda3/envs/pytorch/lib/python3.7/site-packages/torch/testing/_internal/common_utils.py", line 266, in run_tests
    unittest.main(argv=argv)
  File "/home/supriyar/local/miniconda3/envs/pytorch/lib/python3.7/unittest/main.py", line 100, in __init__
    self.parseArgs(argv)
  File "/home/supriyar/local/miniconda3/envs/pytorch/lib/python3.7/unittest/main.py", line 147, in parseArgs
    self.createTests()
  File "/home/supriyar/local/miniconda3/envs/pytorch/lib/python3.7/unittest/main.py", line 159, in createTests
    self.module)
  File "/home/supriyar/local/miniconda3/envs/pytorch/lib/python3.7/unittest/loader.py", line 220, in loadTestsFromNames
    suites = [self.loadTestsFromName(name, module) for name in names]
  File "/home/supriyar/local/miniconda3/envs/pytorch/lib/python3.7/unittest/loader.py", line 220, in <listcomp>
    suites = [self.loadTestsFromName(name, module) for name in names]
  File "/home/supriyar/local/miniconda3/envs/pytorch/lib/python3.7/unittest/loader.py", line 205, in loadTestsFromName
    test = obj()
  File "/home/supriyar/local/miniconda3/envs/pytorch/lib/python3.7/site-packages/torch/testing/_internal/common_quantization.py", line 129, in wrapper
    fn(*args, **kwargs)
  File "/home/supriyar/local/miniconda3/envs/pytorch/lib/python3.7/site-packages/torch/testing/_internal/common_utils.py", line 714, in __init__
    test_method = getattr(self, method_name)
AttributeError: 'TestPostTrainingDynamic' object has no attribute 'runTest'
```

Reply:
@supriyar can you share a bit more details on your build config?

Reply:
Sorry about that, here is the fix: https://github.com/pytorch/pytorch/pull/38432